projects/core/src/routing/url-matcher/default-url-matcher.ts

```diff
@@ -0,0 +1,34 @@
+import { inject, InjectionToken } from '@angular/core';
+import { Route } from '@angular/router';
+import { RoutingConfigService } from '../configurable-routes';
+import { UrlMatcherService } from '../services/url-matcher.service';
+import { UrlMatcherFactory } from './url-matcher-factory';
+
+export function getDefaultUrlMatcherFactory(
+  routingConfigService: RoutingConfigService,
+  urlMatcherService: UrlMatcherService
+): UrlMatcherFactory {
+  const factory = (route: Route) => {
+    const routeName = route.data && route.data['cxRoute'];
+    const routeConfig = routingConfigService.getRouteConfig(routeName);
+    const paths = (routeConfig && routeConfig.paths) || [];
+    return urlMatcherService.getFromPaths(paths);
+  };
+  return factory;
+}
+
+/**
+ * Injection token with url matcher factory for spartacus routes containing property `data.cxRoute`.
+ * The provided url matcher matches the configured `paths` from routing config.
```

Comment:
What about adding a little more doc here:

If the default matcher doesn't fit the requirements, the `DEFAULT_URL_MATCHER` can be replaced or additional matchers can be added for a specific route. See for example [route matcher example].

Matchers will "match" a route, but do not contribute to the creation of the route, nor do they guard routes.